graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+        """
+        Select molecules with a given oligmeric length.
+
+        :param length: Oligomeric length of molecule, defaults to ``1``.
+        :type length: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected oligmers.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.length == oligomer]
+
+        if update:
+            self.df = df
+        return df
+
+    def has_ligand(self, ligand: str, update: bool = False) -> pd.DataFrame:
+        """
+        Select molecules that contain a given ligand.
+
+        :param ligand: Ligand to select. (PDB ligand code)
+        :type ligand: str
+        :param update: Whether to update the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.ligands.map(lambda x: ligand in x)]
+
+        if update:
+            self.df = df
+        return df
+
+    def has_ligands(
+        self, ligands: List[str], inverse: bool = False, update: bool = False
+    ):
+        """Select molecules that contain all ligands in the provided list.
+
+        If inverse is ``True``, selects molecules that do not have all the
+        ligands in the list.
+
+        :param ligand: List of ligands. (PDB ligand codes)
+        :type ligand: List[str]
+        :param inverse: Whether to inverse the selection, defaults to ``False``.
+        :type inverse: bool, optional
+        :param update: Whether to update the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        if inverse:
+            df = self.df.loc[
+                self.df.ligands.map(lambda x: not set(ligands).issubset(x))
+            ]
+        else:
+            df = self.df.loc[
+                self.df.ligands.map(lambda x: set(ligands).issubset(x))
+            ]
+
+        if update:
+            self.df = df
+        return df
+
+    def to_dict(self) -> Dict[str, str]:
+        """Return a dictionary of sequences indexed by chains.
+
+        :return: Dictionary of chain-sequence mappings.
+        :rtype: Dict[str, str]
+        """
+        return (
+            self.df[["id", "sequence"]].set_index("id").to_dict()["sequence"]
+        )
+
+    def to_fasta(self, filename: str):
+        """Write the dataset to a FASTA file (indexed by chain id).
+
+        :param filename: Name of the output FASTA file.
+        :type filename: str
+        """
+        with open(filename, "w") as f:
+            for k, v in self.to_dict().items():
+                f.write(f">{k}\n")
+                f.write(f"{v}\n")
+
+    def standard_alphabet(self, update: bool = False):
```

Comment:
Needs a clearer name

Reply:
Addressed in [this commit](https://github.com/a-r-j/graphein/pull/272/commits/3b42dab1be3429af2e9b2c5da59166bc722c8046).